guides/v2.1/contributor-guide/backward-compatible-development/index.md

```diff
@@ -39,6 +39,12 @@ Mark the method with the `@deprecated` tag instead of removing it.
 
 Continue returning the same results from the method if possible, so the old functionality is preserved.
 
+If you need to remove the existing functionality and there is no alternative implementation, you MUST change the method code to
```

Comment:
Actually, you can **add** `trigger_error` method but don't replace the existing. The method should still return the same result.

Reply:
It is impossible to return the same result and trigger error at the same time. I've changed the type of error

Reply:
What type of error are you trying to use? Doesn't it work with `E_WARNING`?

Reply:
`E_USER_ERROR` is a real PHP error and this avoids hidden warnings in the log

Reply:
@fascinosum why to don't use `E_DEPRECATED`? PHP core team uses it to "mark" deprecated functionality. Why the deprecated code should throw an error?

Reply:
@joni-jones because this is not a deprecated code, but a backward incompatible change. You must not use this method anymore

Reply:
BC guide should not contain approaches which introduce BIC.

Reply:
Why should we keep a method if we can not mimic an execution result?